### PR TITLE
Implemented branch coverage tool for AuthorListParser::getToken

### PR DESCRIPTION
**Pull request by yiiju**
**Added**

* Branch coverage tool for AuthorListParser::getToken.

### CODE REVIEW

File: src/main/java/org/jabref/model/entry/AuthorListParser.java [hunk 1]
```diff
@@ -333,85 +339,188 @@ private String concatTokens(List<Object> tokens, int start, int end, int offset,
...
             }
+            else {
+                visited[3] = true;
+            }
             tokenStart++;
         }
         tokenEnd = tokenStart;
         if (tokenStart >= original.length()) {
+            visited[4] = true;
             return TOKEN_EOF;
```
Comment (SoFoDa) [inline src/main/java/org/jabref/model/entry/AuthorListParser.java:360]: You could move the file writing to a separate function and call it before each return statement in order to make sure that every branch is correctly represented in the coverage file.

**Review by SoFoDa — APPROVED** [review]
Nice work 👍

**Review by christinerosquist — APPROVED** [review]
Looks good 🔥

File: src/main/java/org/jabref/model/entry/AuthorListParser.java (hunk 1 above)
Comment (simonsirak) [reply src/main/java/org/jabref/model/entry/AuthorListParser.java:360]: There are some return statements that need to call `branchCoverage()` beforehand!

**Review by simonsirak — COMMENTED** [review]
Looks good, nice job! I had a similar comment as David but otherwise it looks great!

**Pull request merged**